sdk/state/ingest.go

```diff
@@ -108,3 +115,37 @@ func (c *Channel) ingestTxToUpdateUnauthorizedCloseAgreement(tx *txnbuild.Transa
 
 	return nil
 }
+
+func (c *Channel) ingestTxToUpdateBalances(tx *txnbuild.Transaction) error {
+	localEscrowAddress := c.localEscrowAccount.Address.Address()
+	remoteEscrowAddress := c.remoteEscrowAccount.Address.Address()
+
+	for _, op := range tx.Operations() {
+		if paymentOp, ok := op.(*txnbuild.Payment); ok {
+			amount, err := amount.ParseInt64(paymentOp.Amount)
+			if err != nil {
+				return fmt.Errorf("payment operation amount is not parsable")
+			}
+
+			// Check for a withdrawal. Check operation source first then tx source.
+			if paymentOp.SourceAccount == localEscrowAddress {
+				c.localEscrowAccount.Balance -= amount
+			} else if paymentOp.SourceAccount == remoteEscrowAddress {
+				c.remoteEscrowAccount.Balance -= amount
+			} else if tx.SourceAccount().AccountID == localEscrowAddress {
+				c.localEscrowAccount.Balance -= amount
+			} else if tx.SourceAccount().AccountID == remoteEscrowAddress {
+				c.remoteEscrowAccount.Balance -= amount
+			}
```

Comment:
I think these conditions need to check that the tx source is the account and the op source is nil or equal too, or only the op source is equal. I think by checking each only individually there's an edge case being missed here where the escrow account is the tx source but the op source is some other account. I can't imagine what tx would do that, but we should probably avoid making that assumption in this code.

Reply:
I think I have that test case covered [here](https://github.com/stellar/experimental-payment-channels/pull/234/files#diff-342d9dd35e25c88e1eb70e4d18a848b11d59461fc433e1e802686b6ba3cb73f7R256-R264). I tried covering all the edges cases in that test lmk if I'm missing one though.

My thinking here is first check if the op source is an escrow account. If op source is nil or some other account, then go to the next `else if` looking at the tx source

Reply:
resolving since this is changed now